components/automate-ui/src/app/components/sidebar/sidebar.component.ts

```diff
@@ -13,17 +14,26 @@ export class SidebarComponent {
   menuGroups$: Observable<MenuItemGroup[]>;
 
   constructor(
-    @Inject(LayoutFacadeService) public layoutFacade: LayoutFacadeService
+    @Inject(LayoutFacadeService) public layoutFacade: LayoutFacadeService,
+    private TelemetryService: TelemetryService
   ) {
     this.menuGroups$ = layoutFacade.sidebar$;
     this.updateMenuGroupVisibility();
   }
 
   public isAuthorized($event, menuItem, menuGroup) {
     menuItem.authorized.isAuthorized = $event;
+    this.checkDeploymentType(menuItem , menuGroup)
     this.setGroupVisibility(menuGroup);
   }
 
+  public checkDeploymentType(menuItem: any , menuGroup: any) {
+    if(menuGroup.name == 'Single Sign-On' && menuItem.route == '/settings/sso' && this.TelemetryService.getDeploymenType() != 'SAAS') {
```

Comment:
The value `Single Sign-On` should ideally be imported from a constant and then used as comparison.